lighthouse-core/audits/apple-touch-icon.js

```diff
@@ -44,23 +47,42 @@ class IosPwaIcon extends Audit {
 
   /**
    * @param {LH.Artifacts} artifacts
-   * @param {LH.Audit.Context} _
    * @return {LH.Audit.Product}
    */
-  static audit(artifacts, _) {
+  static audit(artifacts) {
+    let explanation;
     const appleTouchIcons = artifacts.LinkElements.filter(
-      el => el.rel === 'apple-touch-icon' && el.href !== undefined);
+      el => {
+        if (el.rel !== 'apple-touch-icon' && el.rel !== 'apple-touch-icon-precomposed'
+          || !el.href) {
+          return false;
+        }
+        // Check that the href is valid
```

Comment:
this section makes my eyes bleed. 🤕

i prefer the arr.filter().filter().filter() we do elsewhere.

Reply:
lol i can do that instead, but the href stuff is blegh, because it will fail 1 of the 2, so it has to be this kind of garbage somwhere

Reply:
FWIW @exterkamp you shouldn't need to do this at all, `LinkElements` already does the normalization on `href`, just checking if `href` is present should be enough (`hrefRaw` contains the potentially invalid one)

Reply:
I realize it was my comment that was probably misleading with the validation, I just meant checking if `href` was truthy and potentially fetching the url to see if it's a valid icon :)